src/RequestExecutor/HttpRequestExecutor.spec.ts

```diff
@@ -0,0 +1,174 @@
+import 'reflect-metadata';
+import 'chai/register-should';
+import { HttpRequestExecutor } from './HttpRequestExecutor';
+import {
+  DefaultVirtualScripts,
+  VirtualScript,
+  VirtualScripts,
+  VirtualScriptType
+} from '../Scripts';
+import { Protocol } from './Protocol';
+import { RequestExecutor } from './RequestExecutor';
+import { Request, RequestOptions } from './Request';
+import { RequestExecutorOptions } from './RequestExecutorOptions';
+import { expect } from 'chai';
+import nock from 'nock';
+import {
+  anyString,
+  anything,
+  instance,
+  mock,
+  reset,
+  spy,
+  verify,
+  when
+} from 'ts-mockito';
+import { container, Lifecycle } from 'tsyringe';
+import { URL } from 'url';
+
+describe('HttpRequestExecutor', () => {
+  const createRequest = (options?: Partial<RequestOptions>) => {
+    const requestOptions = { url: 'https://foo.bar', headers: {}, ...options };
+    const request = new Request(requestOptions);
+    const spiedRequest = spy(request);
+    when(spiedRequest.method).thenReturn('GET');
+
+    return { requestOptions, request, spiedRequest };
+  };
```

Comment:
It is not clear to me, why did you declared this function inside `describe`?

Reply:
I declared it this way because it only makes sense inside this particular test suite. So generally I think of `describe` blocks as of test suites. If that's not the case in this project (i.e. a **file** is a test suite, and a `describe` block is just a part of it) then I'm totally fine with it and I can move it out.